iceoryx_hoofs/include/iceoryx_hoofs/cxx/helplets.hpp

```diff
@@ -217,153 +67,13 @@ static constexpr uint64_t arrayCapacity(T const (&/*notInterested*/)[CapacityVal
     return CapacityValue;
 }
 
-/// @brief get the best fitting unsigned integer type for a given value at compile time
-template <uint64_t Value>
-struct BestFittingType
-{
-/// ignore the warnings because we need the comparisons to find the best fitting type
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wtype-limits"
-    using Type_t = typename internal::BestFittingTypeImpl<(Value > std::numeric_limits<uint8_t>::max()),
-                                                          (Value > std::numeric_limits<uint16_t>::max()),
-                                                          (Value > std::numeric_limits<uint32_t>::max())>::Type_t;
-#pragma GCC diagnostic pop
-};
-
-template <uint64_t Value>
-using BestFittingType_t = typename BestFittingType<Value>::Type_t;
-
 /// @brief Returns info whether called on a 32-bit system
 /// @return True if called on 32-bit, false if not 32-bit system
 constexpr bool isCompiledOn32BitSystem() noexcept
 {
     return INTPTR_MAX == INT32_MAX;
 }
```

Comment:
Can can this be moved to `system_configuration`? Then the `helplets` would be gone for good.